In [2]:
import pandas as pd
from sklearn.metrics import classification_report

In [3]:
df = pd.read_csv("/Users/jennihawk/Documents/Data Science Projects/Project_NLP/Advanced_NLP_Project/Scoring Intrusion/result_draft_final_annotated.csv")

df.head()

Unnamed: 0                                             Tweets  \
0           0  @The_FJC That #January6thCommitteeHearings was...   
1           1  @washingtonpost #America #AmericaFirst #Januar...   
2           2  Donald Trump knew he lost.” 2/2 - Chmn. Benn...   
3           3  These #January6thCommitteeHearings have been e...   
4           4  RT @ColorMeStevie: Mr. Trump is responsible fo...   

                                    Processed_Tweets       topic_label  \
0  fjc planned yesterday exactly reason knew repo...  Mitch_Kevin_Knew   
1      america americafirst making martyr trump maga  Hamill_Democracy   
2  donald trump knew lost 2 2 chmn bennie thompso...  Mitch_Kevin_Knew   
3  explosive knew jan 6 premeditated tfg amp team...  Mitch_Kevin_Knew   
4           mr trump responsible 6th period end lock  Mitch_Kevin_Knew   

   intruder  intruder_index  jenni_intruder  
0         0             NaN             NaN  
1         1             1.0             1.0  
2         0             NaN             NaN  
3         0             NaN             NaN  
4         0             NaN             NaN

In [6]:
# fill nans with 0 so chris_intruder and intruder look the same
df["jenni_intruder"] = df["jenni_intruder"].fillna(0)

#true data | ground truth
y_true = df["intruder"]

# model predictions - Human is also a model so this is our human prediction
y_pred = df["jenni_intruder"]

In [8]:
df.shape

(60, 7)

In [7]:
print(
    classification_report(
        y_true, y_pred, target_names=["Does not matter", "Doc intrusion"]
    )
)

                 precision    recall  f1-score   support

Does not matter       0.83      0.83      0.83        48
  Doc intrusion       0.33      0.33      0.33        12

       accuracy                           0.73        60
      macro avg       0.58      0.58      0.58        60
   weighted avg       0.73      0.73      0.73        60



### Findings
- How does this model score? 
    - Humans selected the intruder at a rate of 33%. Confidence in how this model is determining coherence is low. 
- Which topics scored best? 